src/conditions/datadog_search.rs

```diff
@@ -0,0 +1,658 @@
+            ),
+            // Facet match (negate).
+            (
+                "NOT @a:bla",
+                log_event!["tags" => vec!["a:bla"]],
+                log_event!["custom" => json!({"a": "bla"})],
+            ),
+            // Facet match (negate w/-).
+            (
+                "-@a:bla",
+                log_event!["tags" => vec!["a:bla"]],
+                log_event!["custom" => json!({"a": "bla"})],
+            ),
+            // Quoted facet match.
+            (
+                r#"@a:"bla""#,
+                log_event!["custom" => json!({"a": "bla"})],
+                log_event!["tags" => vec!["a:bla"]],
+            ),
+            // Quoted facet match (negate).
+            (
+                r#"NOT @a:"bla""#,
+                log_event!["tags" => vec!["a:bla"]],
+                log_event!["custom" => json!({"a": "bla"})],
+            ),
+            // Quoted facet match (negate w/-).
+            (
+                r#"-@a:"bla""#,
+                log_event!["tags" => vec!["a:bla"]],
+                log_event!["custom" => json!({"a": "bla"})],
+            ),
+            // Wildcard prefix.
+            (
+                "*bla",
+                log_event!["message" => "foobla"],
+                log_event!["message" => "blafoo"],
+            ),
+            // Wildcard prefix (negate).
+            (
+                "NOT *bla",
+                log_event!["message" => "blafoo"],
+                log_event!["message" => "foobla"],
+            ),
+            // Wildcard prefix (negate w/-).
+            (
+                "-*bla",
+                log_event!["message" => "blafoo"],
+                log_event!["message" => "foobla"],
+            ),
+            // Wildcard suffix.
+            (
+                "bla*",
+                log_event!["message" => "blafoo"],
+                log_event!["message" => "foobla"],
+            ),
+            // Wildcard suffix (negate).
+            (
+                "NOT bla*",
+                log_event!["message" => "foobla"],
+                log_event!["message" => "blafoo"],
+            ),
+            // Wildcard suffix (negate w/-).
+            (
+                "-bla*",
+                log_event!["message" => "foobla"],
+                log_event!["message" => "blafoo"],
+            ),
+            // Multiple wildcards.
+            (
+                "*b*la*",
+                log_event!["custom" => r#"{"title": "foobla"}"#],
+                log_event![],
+            ),
+            // Multiple wildcards (negate).
+            (
+                "NOT *b*la*",
+                log_event![],
+                log_event!["custom" => r#"{"title": "foobla"}"#],
+            ),
+            // Multiple wildcards (negate w/-).
+            (
+                "-*b*la*",
+                log_event![],
+                log_event!["custom" => r#"{"title": "foobla"}"#],
+            ),
+            // Wildcard prefix - tag.
+            (
```

Comment:
I guess we need to updates these tests after the most recent search syntax changes, right?

Reply:
The match use-case should be something like: `log_event!["tags" => vec!["a:foobla"]]`